src/components/shared/mailchimp/mailchimp.js

```diff
@@ -1,46 +1,82 @@
-import React from 'react'
+import React, { useState, useRef } from 'react'
 import addToMailchimp from 'gatsby-plugin-mailchimp'
 
-class Mailchimp extends React.Component {
-  state = {
-    resultMessage: '',
-    error: false,
-  }
+function Mailchimp() {
+  const [resultMessage, setResultMessage] = useState('')
+  const [error, setError] = useState(false)
+  const [FNAME, setFNAME] = useState('')
+  const [LNAME, setLNAME] = useState('')
+  const [EMAIL, setEMAIL] = useState('')
+
+  const resultMessageClass = error ? 'error-message' : 'success-message'
 
-  handleSubmit = (e) => {
-    e.preventDefault();
-    const formData = new FormData(this.form);
+  const handleSubmit = e => {
+    e.preventDefault()
+    const formData = new FormData(e.target)
     const email = formData.get('EMAIL')
     const fields = {
-        FNAME: formData.get('FNAME'),
-        LNAME: formData.get('LNAME'),
+      FNAME: formData.get('FNAME'),
+      LNAME: formData.get('LNAME'),
     }
-    console.log(email);
-    console.log(fields);
-    addToMailchimp(email, fields).then((result) => {
-      this.setState({
-        resultMessage: result.msg.split('<a')[0],
-        error: result.result === 'error',
-      })
-      this.form.reset();
-    });
+    addToMailchimp(email, fields).then(result => {
+      setResultMessage(result.msg.split('<a')[0])
+      setError(result.result === 'error')
+    })
```

Comment:
I was testing this and it wasn't pulling the form fields into Mailchimp. Instead of using the FormData, we can use the state of the form, which should be more reliable.

```
  const handleSubmit = e => {
    e.preventDefault()
    const email = EMAIL
    const fields = {
      FNAME: FNAME,
      LNAME: LNAME,
    }
    addToMailchimp(email, fields).then(result => {
      setResultMessage(result.msg.split('<a')[0])
      setError(result.result === 'error')
    })
  }
```